chia/server/start_farmer.py

```diff
@@ -36,7 +37,9 @@ def create_farmer_service(
     connect_peers = []
     fnp = service_config.get("full_node_peer")
     if fnp is not None:
-        connect_peers.append(PeerInfo(fnp["host"], fnp["port"]))
+        connect_peers.append(
+            PeerInfo(str(get_host_addr(fnp["host"], prefer_ipv6=config.get("prefer_ipv6", False))), fnp["port"])
```

Comment:
is `get_host_addr()` not `async`? this looks suspicious. If it's a blocking call, we'll block the whole mainloop

Reply:
Its not `async`. https://github.com/Chia-Network/chia-blockchain/blob/cd4874c584046daa9bdf77ed05c37f3d42de0a8a/chia/util/network.py#L112
so yeah blocking but does it matter where exactly we block on startup? I mean, right now it happens in `Service.run` and after this PR its just moved few layers up in the startup process into `create_<whatever>_service`